traits/has_traits.py

```diff
@@ -619,7 +630,23 @@ def update_traits_class_dict(class_name, bases, class_dict):
                 )
 
         handlers = [h for h in handlers if h is not None]
+
         default = _get_def(class_name, class_dict, [], "_%s_default" % name)
+        # In case of mapped traits, need to form a new default method
+        if (name[-1:] == '_'):
```

Comment:
I guess the solution I was thinking of was different: that is, in the branch `if default is not None` below, set both the default value of the main trait _and_ (in the case of a mapped trait) the default value of the shadow trait at the same time. Would that work, or is there something I'm missing?

I'm not too worried about what happens if the user's created a `default` method just for the shadow trait; that seems as though it should almost always be an error (especially if there's _also_ a default for the main trait).

Reply:
(Nitpick: unnecessary parentheses around the `if` condition.)

Reply:
> I guess the solution I was thinking of was different: that is, in the branch `if default is not None` below, set both the default value of the main trait _and_ (in the case of a mapped trait) the default value of the shadow trait at the same time. Would that work, or is there something I'm missing?

It would've worked and would've made the code a bit cleaner compared to the original approach in the draft. Not relevant to the new approach, so marking this as resolved.